### PR TITLE
[serve] Add log file path to replica details

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Add absolute file path to log files for each replica.
https://github.com/ray-project/ray/pull/33503#discussion_r1142835813

Example:
```
replicas:
- replica_id: foo_DAGDriver#jsrUNs
  state: RUNNING
  pid: 68276
  actor_name: SERVE_REPLICA::foo_DAGDriver#jsrUNs
  actor_id: 7c1c702270bb634a7cf4c24f01000000
  node_id: 568bf20e0658e89361a997fe57b896b15fcb97268f3b039e1513c6a5
  node_ip: 192.168.1.14
  start_time_s: 1679598497.387779
  log_file_path_id: /serve/deployment_foo_DAGDriver_foo_DAGDriver#jsrUNs.log
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/replica.py [hunk 1]
```diff
@@ -213,13 +213,20 @@ async def is_allocated(self) -> str:
...
 
             Returns:
-                The PID, actor ID, node ID, node IP of the replica.
+                The PID, actor ID, node ID, node IP, and log file path of the replica.
             """
+
+            log_file_path = None
+            for handler in logger.handlers:
+                if isinstance(handler, logging.handlers.RotatingFileHandler):
+                    log_file_path = handler.baseFilename
```
Comment (edoakes) [inline python/ray/serve/_private/replica.py:222]: this logic should live in `logging_utils` along with the logic that sets the file path

**Review by edoakes — CHANGES_REQUESTED** [review]

File: python/ray/serve/_private/logging_utils.py [hunk 2]
```diff
@@ -47,6 +47,13 @@ def access_log_msg(*, method: str, status: str, latency_ms: float):
     return f"{method.upper()} {status.upper()} {latency_ms:.1f}ms"
 
 
+def get_logger_file_path() -> Optional[str]:
```
Comment (edoakes) [inline python/ray/serve/_private/logging_utils.py:50]:
```suggestion
def get_component_logger_file_path() -> Optional[str]:
```
nit: match name below

**Review by edoakes — APPROVED** [review]

**Comment by zcin** [comment]
May need to change file path from absolute to relative file path per @alanwguo's request

**Comment by edoakes** [comment]
I think this should definitely be an absolutely path, what's the purpose of making it relative? I don't know what it would even be relative to...

File: python/ray/serve/schema.py [hunk 3]
```diff
@@ -634,6 +634,12 @@ class ReplicaDetails(BaseModel, extra=Extra.forbid, frozen=True):
             "state from the running replica actor."
         )
     )
+    log_file_path_id: Optional[str] = Field(
```
Comment (edoakes) [inline python/ray/serve/schema.py:637]:
```suggestion
    log_file_path: Optional[str] = Field(
```

**Review by edoakes — CHANGES_REQUESTED** [review]

**Review by edoakes — APPROVED** [review]

**Comment by zcin** [comment]
@edoakes Tests look good, ready for merge

**Pull request merged**